### PR TITLE
CDAP-3356 remove deprecated 'id' field from RESTful APIs

### PR DESCRIPTION
**Pull request by albertshau**
More specifically, removing from the apps and programs RESTful
APIs. Id was deprecated because it always had the same value as
name. Also removing unused classes for streams and datasets.

### CODE REVIEW

File: cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java [hunk 1]
```diff
@@ -274,7 +273,7 @@ public void setTTL(Id.Stream stream, long ttlInSeconds)
   /**
    * Lists all streams.
    *
-   * @return list of {@link StreamRecord}s
+   * @return list of {@link StreamDetail StreamDetails}
```
Comment (bdmogal) [inline cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java]: is this valid syntax?

**Comment by bdmogal** [comment]
1 question, otherwise LGTM

File: cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java (hunk 1 above)
Comment (chtyim) [inline cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java]: it is.

Comment (awholegunch) [inline cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java]: One is the link, the second is a label. As you can see, lets one use consistent pluralization.

**Comment by awholegunch** [comment]
LGTM. Please copy your release note comment to the new "Release Notes" field added to the Jiras database. That way we will know it has been written up.

**Pull request merged**